src/storage/txn/sched_pool.rs

```diff
@@ -66,6 +70,13 @@ pub enum SchedPool {
         worker_pool: FuturePool,
         resource_ctl: Arc<ResourceController>,
     },
+    // automatically switch between the `single-queue pool` and `priority-queue pool` based on the
+    // resource group settings only used when the resource control feature is enabled.
+    Dynamic {
```

Comment:
Using a nested `Arc<SchedPool>` looks a bit weird here,   I prefer to define this struct as one of the following two:
1. still use enum but define both vanilla and priority as struct:
```
struct VanillaQueue {
 high_worker_pool: FuturePool,
 worker_pool: FuturePool,
}
struct PriorityQueue {
  worker_pool: FuturePool,
   resource_ctl: Arc<ResourceController>,
}
enum SchedPool {
  Vanilla(VanillaQueue),
  Priority(PriorityQueue),
  Dynamic(VanillaQueue, PriorityQueue),
}
``` 
2. Always initialize both of the two queue and use a flag to determine the queue type:
```
enum QueueType {
  Vanilla,
  Priority,
  Dynamic
}

struct SchedPool {
  vanilla: VanillaQueue,
  priority: PriorityQuque,
  type: QueueType,
}
````
This approach can make the code much simplier, then only drawback is that we will always init both pools, but I think this isn't a big problem here.